sdk/core/System.ClientModel/src/Convenience/ClientAuthenticationMethod.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.ClientModel.Primitives;
+/// <summary>
+/// Specifies the kind of connection used by the client.
+/// </summary>
+public enum ClientAuthenticationMethod
+{
+    /// <summary>
+    /// Represents a connection using Entra ID.
+    /// </summary>
+    EntraId,
```

Comment:
What happens when they rebrand this again in the future?   Should we make this something more generic like "Identity" or "Credential"?

Reply:
Or "Credential". This enum value basically means you should be using ClientConnection.Credentail property. BTW, the Credential property is now typed as object, because we don't have a representation of credential in SCM (relevant to our discussion today). The idea is that extension methods that create clients will downcast.

Reply:
Rename `EntraId` to `Credential` and created an issue for the type - https://github.com/Azure/azure-sdk-for-net/issues/48296